lib/temporal/worker.rb

```diff
@@ -0,0 +1,71 @@
+require 'temporal/sdk/core/core_interface_pb'
+require 'temporal/bridge'
+require 'temporal/runtime'
+require 'temporal/worker/reactor'
+
+module Temporal
+  class Worker
+    class AlreadyStarted < StandardError; end
+
+    def initialize(connection, namespace, task_queue)
+      runtime = Temporal::Runtime.instance
+      runtime.ensure_callback_loop
+
+      @running = false
+      @core_worker = Temporal::Bridge::Worker.create(
+        runtime.core_runtime,
+        connection.core_connection,
+        namespace,
+        task_queue
+      )
+    end
+
+    def run
+      start
+      Reactor.attach
+    end
+
+    def start
+      raise AlreadyStarted if running?
+
+      @running = true
+
+      Reactor.execute do |reactor|
+        while running?
+          task = reactor.await { |r| poll_activity_task(r) }
+          reactor.async { |r| process_activity_task(task, r) }
+        end
+      end
+    end
+
+    def shutdown
+      @running = false
```

Comment:
I assume the plan is to send a shutdown to core when this happens?

Reply:
Yes, that's exactly right